base/abstractarray.jl

```diff
@@ -2434,9 +2441,8 @@ function hash(A::AbstractArray, h::UInt)
     n = 0
     while true
```

Comment:
Why not also:
```suggestion
    @inbounds while true
```

I vaguely remember skipping `@inbounds` when we first implemented this because it felt fancy... but we've never gotten out of bounds reports.